Packages/com.unity.inputsystem/Tests/InputSystem/Plugins/UITests.cs

```diff
@@ -94,17 +103,164 @@ public void TDO_Actions_CanDriveUI()
+        Assert.That(leftChildReceiver.events[0].type, Is.EqualTo(EventType.Exit));
+        Assert.That(leftChildReceiver.events[1].type, Is.EqualTo(EventType.Dragging));
+        leftChildReceiver.Reset();
+        Assert.That(rightChildReceiver.events, Has.Count.EqualTo(1));
+        Assert.That(rightChildReceiver.events[0].type, Is.EqualTo(EventType.Enter));
+        rightChildReceiver.Reset();
+
+        // Release button
+        InputSystem.QueueStateEvent(mouse, new MouseState { position = new Vector2(400, 200), buttons = (ushort)0 });
+        InputSystem.Update();
+        eventSystem.InvokeUpdate();
+
+        Assert.That(leftChildReceiver.events, Has.Count.EqualTo(2));
+        Assert.That(leftChildReceiver.events[0].type, Is.EqualTo(EventType.Up));
+        Assert.That(leftChildReceiver.events[1].type, Is.EqualTo(EventType.EndDrag));
+        leftChildReceiver.Reset();
+        Assert.That(rightChildReceiver.events, Has.Count.EqualTo(1));
+        Assert.That(rightChildReceiver.events[0].type, Is.EqualTo(EventType.Drop));
+        rightChildReceiver.Reset();
+
+        // Check Scroll
+        InputSystem.QueueDeltaStateEvent(mouse.scroll, Vector2.one);
+        InputSystem.Update();
+        eventSystem.InvokeUpdate();
+
+        Assert.That(leftChildReceiver.events, Has.Count.EqualTo(0));
+        Assert.That(rightChildReceiver.events, Has.Count.EqualTo(1));
+        Assert.That(rightChildReceiver.events[0].type, Is.EqualTo(EventType.Scroll));
+        rightChildReceiver.Reset();
+
+        // Reset Scroll
+        InputSystem.QueueDeltaStateEvent(mouse.scroll, Vector2.zero);
+        InputSystem.Update();
+        eventSystem.InvokeUpdate();
+
+        Assert.That(leftChildReceiver.events, Has.Count.EqualTo(0));
+        Assert.That(rightChildReceiver.events, Has.Count.EqualTo(1));
+        Assert.That(rightChildReceiver.events[0].type, Is.EqualTo(EventType.Scroll));
+        rightChildReceiver.Reset();
+
+        // Test Selection
+        eventSystem.SetSelectedGameObject(leftChildGameObject);
+        eventSystem.InvokeUpdate();
+
+        Assert.That(leftChildReceiver.events, Has.Count.EqualTo(1));
+        Assert.That(leftChildReceiver.events[0].type, Is.EqualTo(EventType.Select));
+        leftChildReceiver.Reset();
+        Assert.That(rightChildReceiver.events, Has.Count.EqualTo(0));
+
+        // Check Move Axes
+        InputSystem.QueueDeltaStateEvent(gamepad.leftStick, new Vector2(1.0f, 0.0f));
+        InputSystem.Update();
+        eventSystem.InvokeUpdate();
+
+        Assert.That(leftChildReceiver.events, Has.Count.EqualTo(1));
+        Assert.That(leftChildReceiver.events[0].type, Is.EqualTo(EventType.Move));
+        leftChildReceiver.Reset();
+        Assert.That(rightChildReceiver.events, Has.Count.EqualTo(0));
+
+        // Check Submit
+        InputSystem.QueueStateEvent(gamepad, new GamepadState { buttons = (uint)(1 << (int)GamepadState.Button.South) });
```

Comment:
There's a helper for buttons. `new GamepadState().WithButton(GamepadState.Button.South)`. I'm pulling GamepadState.Button out to the toplevel so that it bcomes `new GamepadState().WithButton(GamepadButton.South)`.